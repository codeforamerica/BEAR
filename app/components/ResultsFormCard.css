.sidePadding {
  padding-left: 4em;
  padding-right: 4em;
}

.resultsHeader {
  font-weight: lighter !important;
}

.resultsSubheader {
  font-weight: lighter !important;
  font-size: 1.9rem;
  width: 60%;
  margin-top: 2em;
  margin-left: auto;
  margin-right: auto;
  text-align: left;
}

.subSectionTitle {
  font-weight: 400;
}

.sectionDivider {
  margin: 0 -38px auto;
  width: 960px;
  border-bottom: 2px solid #f7f5f4;
}

.impactSection {
  text-align: center;
  margin-bottom: 20px;
}

.contentsListContainer {
  width: 50%;
  margin: 0 auto;
}

.contentsList {
  list-style: decimal;
  padding-left: 110px;
  margin-top: 0;
  line-height: 1em;
}

.contentsTableTitle {
  margin-top: 2em;
  padding-left: 90px;
  text-align: left;
}

.contentsTable {
  border-bottom: none !important;
  margin: 0 -38px auto;
  width: 960px !important;
}

.contentsTable th {
  padding-right: 15px;
  border-bottom: 2px solid #f7f5f4 !important;
}

.contentsTable td {
  padding-right: 15px;
  border-bottom: none !important;
}

.fileColumn {
  width: 250px;
}

.columnsColumn {
  width: 185px;
}

.rowsColumn {
  width: 170px;
}

.supportingColumn {
  width: 160px;
}

.supportingCheck {
  padding-right: 35px !important;
}

.eligibilityColumn {
  width: 160px;
}

.eligibilityCheck {
  padding-right: 72px !important;
}

.completedIcon {
  background-color: #009d86;
  border-radius: 50%;
  display: inline-block;
  width: 60px;
  height: 60px;
  background-image: url('../assets/images/white_check_large.png');
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 70%;
}

.tealCheck {
  background-image: url('../assets/images/teal_check.png');
}

.resultsFooter {
  margin-top: 0 !important;
}
