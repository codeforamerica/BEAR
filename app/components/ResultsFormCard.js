// @flow
import React, { Component } from 'react';

import FormCard, {
  FormCardContent,
  FormCardFooter,
  FormCardHeader
} from './FormCard';

type Props = {
  county: County,
  outputFolder: string,
  openFolder: string => void
};

export default class ResultsFormCard extends Component<Props> {
  openResultsFolder = () => {
    const { openFolder, outputFolder } = this.props;
    openFolder(outputFolder);
  };

  render() {
    const { county } = this.props;
    return (
      <FormCard>
        <FormCardHeader>
          <div className="text--centered">
            Your files are ready!
            <p>{`We have generated results for ${county.name} county`}</p>
            <button
              className="button button--primary"
              type="button"
              id="view_results"
              onClick={this.openResultsFolder}
            >
              Open Folder
            </button>
          </div>
        </FormCardHeader>
        <FormCardContent />
        <FormCardFooter />
      </FormCard>
    );
  }
}
