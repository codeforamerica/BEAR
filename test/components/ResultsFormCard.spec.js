import sinon from 'sinon';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ResultsFormCard from '../../app/components/ResultsFormCard';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.createSandbox();

function setup() {
  const fakeOpenFolder = sandbox.spy();
  const component = mount(
    <ResultsFormCard
      county="Alameda"
      outputFolder="/path/to/output"
      openFolder={fakeOpenFolder}
    />
  );
  return {
    component,
    fakeOpenFolder
  };
}

afterEach(() => {
  sandbox.restore();
});

describe('ResultsFormCard component', () => {
  describe('clicking the Open Folder button', () => {
    it('should call the openResultsFolder function', () => {
      const { component, fakeOpenFolder } = setup();
      const openFolderButton = component.find('#view_results').at(0);
      openFolderButton.simulate('click');
      expect(fakeOpenFolder.called).toBe(true);
      const { args } = fakeOpenFolder.getCall(0);
      expect(args[0]).toEqual('/path/to/output');
    });
  });

  it('should match exact snapshot', () => {
    const component = (
      <div>
        <ResultsFormCard county="Alameda" />
      </div>
    );

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
