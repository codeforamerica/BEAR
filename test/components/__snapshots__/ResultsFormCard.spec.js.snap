// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFormCard component should match exact snapshot 1`] = `
<div>
  <div
    className="form-card"
  >
    <header
      className="form-card__header"
    >
      <h1
        className="form-card__title"
      >
        <div
          className="text--centered"
        >
          Your files are ready!
          <p>
            We have generated results for undefined county
          </p>
          <button
            className="button button--primary"
            id="view_results"
            onClick={[Function]}
            type="button"
          >
            Open Folder
          </button>
        </div>
      </h1>
    </header>
    <div
      className="form-card__content"
    />
    <div
      className="form-card__footer"
    />
  </div>
</div>
`;
